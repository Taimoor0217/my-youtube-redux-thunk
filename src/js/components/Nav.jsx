import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Search from './Search';
import { getVideos } from '../actions';

const Nav = (props) => {
  return (
    <div className="nav">
      <Search {...props} />
    </div>
  )
}

Nav.propTypes = {
  getVideos: PropTypes.func.isRequired,
}

export default connect(null, {getVideos})(Nav);