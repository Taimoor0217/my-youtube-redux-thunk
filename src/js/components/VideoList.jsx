import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import VideoListEntry from './VideoListEntry';
import { selectVideo } from '../actions';

class VideoList extends Component {

  createList() {
    let { videoList, selectVideo } = this.props;
    return videoList.map((item, i) => {
      return <VideoListEntry key={i} video={item} handleSelectedVideo={selectVideo} />
    })
  }

  render() {
    let { videoList, isLoading, error } = this.props;
    return (
      <div className="videoList">
        {
          isLoading ? <div>Loding Videos....</div> :
          error ? <div>{error.message}</div> : this.createList()
        }
      </div>
    );
  }
}

VideoList.propTypes = {
  videoList: PropTypes.array.isRequired,
  selectVideo: PropTypes.func.isRequired,
}

function mapStateToProps({videoList}) {
  return {
    videoList: videoList.videos,
    isLoading: videoList.isLoading,
    error: videoList.error
  }
};

export default connect(mapStateToProps, {selectVideo})(VideoList);