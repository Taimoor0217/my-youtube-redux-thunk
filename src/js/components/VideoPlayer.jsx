import React, { Component } from 'react';
import PropTypes from 'prop-types';


const VideoPlayer = ({ seletedVideo }) => {
  if(Object.keys(seletedVideo).length) {
    var seletedVideoURL = "https://www.youtube.com/embed/" + seletedVideo.id.videoId
  }
  return (
    <div className="videoPlayer">
      <div className="player embed-responsive embed-responsive-16by9">
        <iframe 
          className="embed-responsive-item"
          src={selectedVideoURL} 
          frameborder="0"
          allowFullScreen
        />
      </div>
      <div className="video-details">
        <h4 className="video-title">
          {Object.keys(selectedVideo).length ? selectedVideo.snippet.title : ""}
        </h4>
        <p className="video-descriptions">
          {Object.keys(seletedVideo).length ? selectedVideo.snippet.description : ""}
        </p>
      </div>
    </div>
  )
};

VideoPlayer.propTypes = {
}

export default VideoPlayer;