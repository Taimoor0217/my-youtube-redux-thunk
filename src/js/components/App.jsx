import React, { Component } from 'react';
import { connect } from 'react-redux';

import Nav from './Nav';
import VideoPlayer from './VideoPlayer';
import VideoList from './VideoList';
import { getVideos } from '../actions';

class App extends Component {
  render() {
    return (
      <div id="app">
        <Nav />
        <div className="col-md-7">
          <VideoPlayer />
        </div>
        <div className="col-md-5">
          <VideoList />
        </div>
      </div>
    );
  }

  componentDidMount() {
    const { getVideos } = this.props;
    getVideos('javascript', true);
  }
}

export default connect(null, {getVideos})(App);