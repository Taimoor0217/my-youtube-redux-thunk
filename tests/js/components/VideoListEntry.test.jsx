import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { VideoList } from '../../../src/js/components/VideoList';

Enzyme.configure({ adapter: new Adapter() });

describe('VideoListEntry Component', () => {
  const videoList = [{}, {}];
  const selectVideo = jest.fn();
  it('renders the VideoList component', () => {
    const isLoading = false;
    const error = null;
    const wrapper = shallow(<VideoList selectVideo={selectVideo} videoList={videoList} isLoading={isLoading} error={error} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the Loading message', () => {
    const isLoading = true;
    const error = null;
    const wrapper = shallow(<VideoList selectVideo={selectVideo} videoList={videoList} isLoading={isLoading} error={error} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the Error message', () => {
    const isLoading = false;
    const error = { message: 'Error while Loading videos' };
    const wrapper = shallow(<VideoList selectVideo={selectVideo} videoList={videoList} isLoading={isLoading} error={error} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('passes on the selecVideo', () => {
    const selectVideo = jest.fn();
    expect(selectVideo.mock.calls.length).toBe(0);
  });
});