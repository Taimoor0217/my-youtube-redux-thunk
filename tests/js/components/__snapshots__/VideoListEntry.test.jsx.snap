// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing VideoList component renders the Error message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={
      Object {
        "message": "Error while Loading videos",
      }
    }
    isLoading={false}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {},
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Error while Loading videos
      </div>,
      "className": "videoList",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Error while Loading videos",
      },
      "ref": null,
      "rendered": "Error while Loading videos",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Error while Loading videos
        </div>,
        "className": "videoList",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error while Loading videos",
        },
        "ref": null,
        "rendered": "Error while Loading videos",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`testing VideoList component renders the Loading message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={null}
    isLoading={true}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {},
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Loading Videos...
      </div>,
      "className": "videoList",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading Videos...",
      },
      "ref": null,
      "rendered": "Loading Videos...",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Loading Videos...
        </div>,
        "className": "videoList",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading Videos...",
        },
        "ref": null,
        "rendered": "Loading Videos...",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
